### DISCUSSION TITLE
Material-UI TextField required validation not working properly

### DISCUSSION THREAD

**Discussion by geekashu** [react-hook-form/react-hook-form#2357] [Q&A] [answered]
I am trying to use Material UI TextField with this library and facing few issues in validation.
Code Sandbox - https://codesandbox.io/s/stupefied-geekashu-4rzb5-4rzb5

I have used two fields, `username` and `password`; both the fields has validations associated with them. 
When you fill the fields with a string and remove them; afterwards you change the focus(tab to another field); nothing happens; ideally the validation should trigger and turn the field to red(error). 

Once you submit the form and try again the same thing; the form will start to behave properly.

How can I make these TextFields to behave from the very beginning, i.e before the form has been submitted for the first time.

Any help is highly appreciated.
Thanks

**Comment by bluebill1049** [comment, marked as answer]
Take a look this: https://www.tutorialspoint.com/Why-do-we-use-the-novalidate-attribute-in-HTML

also prefer to use Controller with MUI

**Reply by geekashu** [reply]
```
  const { register, handleSubmit, errors, formState } = useForm({
    mode: "all"
  });
```
Using mode: "all" fixed the issue. Thanks.

**Comment by suzanamelo-m** [comment]
I'm having the same problem. Also, I have different validations for my input.

const AddPost = ({ addPost }) => {
  const { register, handleSubmit, errors, control } = useForm({
    mode: 'all',
  });

     <form className={classes.form} onSubmit={handleSubmit(onSubmit)}>
            <Controller
              as={TextField}
              control={control}
              variant='outlined'
              margin='normal'
              required
              fullWidth
              id='title'
              label='Title'
              name='title'
              autoFocus
              innerRef={register({
                required: true,
                maxLength: {
                  value: 20,
                  message: 'Your title cannot exceed 30 characters',
                },
                minLength: {
                  value: 2,
                  message: 'Your title must be at least 2 letters',
                },
              })}
              error={errors.title}
              helperText={errors.title ? errors.title.message : ""}
              defaultValue=''
            />

**Reply by suzanamelo-m** [reply]
It just ignores hook-form validations

**Reply by bluebill1049** [reply]
`<form novalidate>` novalidate is what you after.

**Reply by suzanamelo-m** [reply]
I'm afraid I'm forgetting something else. I've tried with and without noValidate, both cases the form still accepting the data without validate as defined inside register...

**Reply by bluebill1049** [reply]
can you provide a codesandbox?

**Reply by suzanamelo-m** [reply]
Is this one:
https://codesandbox.io/s/nice-meadow-qfzgv?file=/src/AddPost.jsx
Thank you so much in advance. I think I'm already blind about it...

**Reply by bluebill1049** [reply]
<img width="1309" alt="Screen Shot 2020-08-17 at 10 00 36 am" src="https://user-images.githubusercontent.com/10513364/90346958-81fcaa80-e070-11ea-8f73-8ef2fd9163b9.png">

looks like it's empty. nothing on the screen.

**Reply by suzanamelo-m** [reply]
Ops! I'm sorry.
https://codesandbox.io/s/hookform-materialui-qfzgv?file=/src/AddPost.jsx

**Comment by bluebill1049** [comment]
haha so simple @suzanamelomoraes 🙏 

you shouldn't need to register inside Controller, for validation use `rules` prop

<img width="1169" alt="Screen Shot 2020-08-17 at 11 15 49 am" src="https://user-images.githubusercontent.com/10513364/90348899-06ecc180-e07b-11ea-96c1-d9a00223f720.png">

**Reply by bluebill1049** [reply]
have a read this section: https://react-hook-form.com/api#Controller
